### ISSUE TITLE
Default route altered

### ISSUE THREAD

**Issue by bvk007** [fabianishere/udm-iptv#38]
Hi,

I've just updated to version 2.0.2. And in this version the default route is updated and internet stops working after restarting igmp proxy until I re-add my default gw route.

cheers, Bas

**Comment by fabianishere** [comment]
What is your ISP? You should check whether the following DHCP options work for you: `-o -O subnet -O broadcast -O staticroutes`.

**Comment by bvk007** [comment]
Glasnet...in the Podman setup I've changed this as well, how do I alter the DHCP options in the new setup?

**Comment by fabianishere** [comment]
The question is shown when running:
```bash
dpkg-reconfigure udm-iptv
```

**Comment by bvk007** [comment]
Thnx

**Comment by bvk007** [comment]
"-o staticroutes" seems to work, with all options (-o -O subnet -O broadcast -O staticroutes) I've got an error

```
● udm-iptv.service - IPTV support for the UniFi Dream Machine
   Loaded: loaded (/lib/systemd/system/udm-iptv.service; enabled; vendor preset: enabled)
   Active: failed (Result: exit-code) since Tue 2022-01-18 15:59:53 CET; 41s ago
  Process: 16754 ExecStart=/usr/lib/udm-iptv/udm-iptv start (code=exited, status=255/EXCEPTION)
 Main PID: 16754 (code=exited, status=255/EXCEPTION)
      CPU: 169ms

Jan 18 15:59:53 ubnt udm-iptv[16754]: ip: SIOCGIFFLAGS: No such device
Jan 18 15:59:53 ubnt udm-iptv[16754]: Obtaining IP address for VLAN interface
Jan 18 15:59:53 ubnt udm-iptv[16754]: udhcpc: SIOCGIFINDEX: No such device
Jan 18 15:59:53 ubnt udm-iptv[16754]: NATing IPTV network ranges (if necessary)
Jan 18 15:59:53 ubnt udm-iptv[16754]: Setting up igmpproxy
Jan 18 15:59:53 ubnt udm-iptv[16754]: Starting igmpproxy
Jan 18 15:59:53 ubnt igmpproxy[16754]: There must be at least 1 Vif as upstream.
Jan 18 15:59:53 ubnt systemd[1]: udm-iptv.service: Main process exited, code=exited, status=255/EXCEPTION
Jan 18 15:59:53 ubnt systemd[1]: udm-iptv.service: Failed with result 'exit-code'.
Jan 18 15:59:53 ubnt systemd[1]: udm-iptv.service: Consumed 169ms CPU time.```

Do I need the other options?

**Comment by fabianishere** [comment]
If that works for you, I would recommend just using that. Otherwise, share the output of `udm-iptv-diag`.

**Comment by bvk007** [comment]
Ok for entertaining purposes: 

```
=== Configuration ===
WAN Interface: eth4
WAN Ranges: 217.166.224.0/22
WAN VLAN: 37
WAN DHCP options: -o -O subnet -O broadcast -O staticroutes
LAN Interfaces: br10
IGMP Proxy quickleave disabled: false
IGMP Proxy debug: true
=== IP Link and Route ===
/usr/bin/udm-iptv-diag: 31: /usr/bin/udm-iptv-diag: ip: not found
/usr/bin/udm-iptv-diag: 32: /usr/bin/udm-iptv-diag: ip: not found
=== Service Logs ===
Jan 18 16:36:42 ubnt udm-iptv[23184]: Obtaining IP address for VLAN interface
Jan 18 16:36:43 ubnt udm-iptv[23184]: udhcpc: SIOCGIFINDEX: No such device
Jan 18 16:36:43 ubnt udm-iptv[23184]: NATing IPTV network ranges (if necessary)
Jan 18 16:36:43 ubnt udm-iptv[23184]: Setting up igmpproxy
Jan 18 16:36:43 ubnt udm-iptv[23184]: Starting igmpproxy
Jan 18 16:36:43 ubnt udm-iptv[23184]: adding VIF, Ix 0 Fl 0x0 IP 0x010aa8c0 br10, Threshold: 1, Ratelimit: 0
Jan 18 16:36:43 ubnt udm-iptv[23184]: There must be at least 1 Vif as upstream.
Jan 18 16:36:43 ubnt systemd[1]: udm-iptv.service: Main process exited, code=exited, status=255/EXCEPTION
Jan 18 16:36:43 ubnt systemd[1]: udm-iptv.service: Failed with result 'exit-code'.
Jan 18 16:36:43 ubnt systemd[1]: udm-iptv.service: Consumed 170ms CPU time.```

**Comment by fabianishere** [comment]
Could you also share the contents of `/etc/udm-iptv.conf`?

**Comment by bvk007** [comment]
```
## Example configuration for udmp-iptv ##
# Interface on which IPTV traffic enters the router
IPTV_WAN_INTERFACE="eth4"
# ID of VLAN which carries IPTV traffic (use 0 if no VLAN is used)
IPTV_WAN_VLAN="37"
# Name of the IPTV VLAN interface
IPTV_WAN_VLAN_INTERFACE="iptv"
# IP ranges from which the IPTV traffic originates (separated by spaces)
IPTV_WAN_RANGES="217.166.224.0/22"
#IPTV_WAN_RANGES="0.0.0.0/0"

# DHCP options to send when requesting an IP address
IPTV_WAN_DHCP_OPTIONS="-o -O subnet -O broadcast -O staticroutes"
# LAN interfaces on which IPTV should be made available
IPTV_LAN_INTERFACES="br10"
# Disable quickleave for igmpproxy
IPTV_IGMPPROXY_DISABLE_QUICKLEAVE="false"
# Enable debugging for igmpproxy
IPTV_IGMPPROXY_DEBUG="false"
```

**Comment by fabianishere** [comment]
Which device are you running this on? I am unable to reproduce the issue you are facing.

**Comment by bvk007** [comment]
UDM (base)

**Comment by bvk007** [comment]
Hmm looks like the service is a bit "wonky" today. Now with the DHCP options the proxy starts

**Comment by fabianishere** [comment]
The issue should be resolved with 3bcca8d.
